packages/medusa/src/interfaces/csv-parser.ts

```diff
@@ -40,21 +40,28 @@ export abstract class AbstractCsvValidator<TCsvLine, TBuiltLine>
   ): Promise<boolean | never>
 }
 
-export type CsvSchemaColumn<TCsvLine, TBuiltLine> = {
-  name: string
+export type CsvSchemaColumn<
+  TCsvLine,
+  TBuiltLine,
+  NameAsOptional = false
+> = (NameAsOptional extends false
+  ? {
+      name: string
+    }
+  : {}) & {
```

Comment:
No it is the other way around, if you pass `true` as the `NameAsOptional` template, then the name property is removed (I'll change that to just make it optional instead). The default value is false and therefore the name is required and merged with the other types.
`Something extends false` means that typescript will check that the value of `Something` is false since the only way to extend false is false itself :)